src/DependencyInjection/BaldinofRoadRunnerExtension.php

```diff
@@ -239,10 +269,178 @@ private function configureKv(array $config, ContainerBuilder $container): void
+                        'client_key' => $options['client_key'] ?? null,
+                        'client_pem' => $options['client_pem'] ?? null,
+                        'override_server_name' => $options['override_server_name'] ?? null,
+                    ],
+                ]);
+
+            $container->register("temporal.client.{$name}.service_client.factory", ServiceClientFactory::class)
+                ->setArguments([
+                    '$connection' => new Reference("temporal.client.{$name}.connection"),
+                ]);
+
+            $container->register("temporal.client.{$name}.service_client", ServiceClientInterface::class)
+                ->setFactory([new Reference("temporal.client.{$name}.service_client.factory"), '__invoke'])
+                ->setAutoconfigured(true)
+                ->setAutowired(true);
+
+            $container->register("temporal.client.{$name}.option", ClientOptions::class)
+                ->setFactory([ClientOptionsFactory::class, 'createFromArray'])
+                ->setArguments([
+                    '$options' => [
+                        'namespace' => $options['namespace'],
+                        'identity' => $options['identity'] ?? null,
+                        'query_reject_condition' => $options['query_reject_condition'] ?? null,
+                    ],
+                ]);
+
+            $container->register("temporal.client.{$name}.interceptors", SimplePipelineProvider::class)
+                ->addArgument(array_map(service(...), $options['interceptors'] ?? []));
+
+            $container->register("temporal.client.{$name}.factory", WorkflowClientFactory::class)
+                ->setArguments([
+                    '$serviceClient' => new Reference("temporal.client.{$name}.service_client"),
+                    '$dataConverter' => new Reference('temporal.data_converter'),
+                    '$clientOptions' => new Reference("temporal.client.{$name}.option"),
+                    '$interceptors' => new Reference("temporal.client.{$name}.interceptors"),
+                ]);
+
+            $container->register("temporal.client.{$name}", WorkflowClient::class)
+                ->setFactory([new Reference("temporal.client.{$name}.factory"), '__invoke'])
+                ->setAutoconfigured(true)
+                ->setAutowired(true);
+
+            $container->register("temporal.client.{$name}.schedule.factory", ScheduleClientFactory::class)
+                ->setArguments([
+                    '$serviceClient' => new Reference("temporal.client.{$name}.service_client"),
+                    '$dataConverter' => new Reference('temporal.data_converter'),
+                    '$clientOptions' => new Reference("temporal.client.{$name}.option"),
+                ]);
+
+            $container->register("temporal.client.{$name}.schedule", ScheduleClient::class)
+                ->setFactory([new Reference("temporal.client.{$name}.schedule.factory"), '__invoke'])
```

Comment:
this schedule factory can be removed so that schedule client creation would be much more straightforward

```suggestion
            $container->register("temporal.client.{$name}.schedule", \Temporal\Client\ScheduleClientInterface::class)
                ->setFactory([\Temporal\Client\ScheduleClient::class, 'create'])
                ->setArguments([
                    '$serviceClient' => new Reference("temporal.client.{$name}.service_client"),
                    '$options' => new Reference("temporal.client.{$name}.option"),
                    '$converter' => new Reference('temporal.data_converter'),
                ]);
```